PYIC-7876: Remove SsmConfigService

MikeCollingwood opened the pull request:
## Proposed changes

### What changed

- Cleanup SSM

### Why did it change

- AppConfig fills the config requirements now

### Issue tracking

- [PYIC-7876](https://govukverify.atlassian.net/browse/PYIC-7876)

## Checklists

- [x] All READMEs and documentation updated where necessary
- [x] No risk of PII, credentials or anything else sensitive being exposed through logs
- [x] API/unit/contract tests have been written/updated
- [x] Production changes appropriately staged out


[PYIC-7876]: https://govukverify.atlassian.net/browse/PYIC-7876?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-7876: Cleanup SSM

DanCorderIPV commented on libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:
```diff
@@ -105,14 +124,14 @@ public List<String> getStringListParameter(
 
+    public abstract List<String> getFeatureSet();
+
+    protected abstract String getSecret(String path);
```
I find it nice to put abstract methods near the top of the class so that people that want to inherit from the class can easily see what they are going to need to implement.

MikeCollingwood replied:
Thank you, that makes sense

MikeCollingwood committed:
PYIC-8506: Fix Sonarcloud issues and move abstract methods to top of abstract class

MikeCollingwood committed:
PYIC-7876:
- Rename YamlConfigService => LocalConfigService
- Added items added to YamlParametersConfigService to ConfigService after lost in merge

MikeCollingwood committed:
PYIC-7876: Fix sonarcloud issues

undermad committed:
PYIC-7876 Rebase main and resolve conflicts.
